### ISSUE TITLE
Potion effect still work after death

### ISSUE THREAD

**Issue by amadin** [minetest-LOTR/Lord-of-the-Test#78]
Is normal if potions effects still work after death?
Effects of the same potion are summarizes if use more than 1 the same potion.

**Comment by Wuzzy2** [comment]
Confirmed. I am very sure this is not normal. :D

In fact, potion effects seem to be completely broken in LOTT. Just drink many different potions in quick succession and wait for the effects to wear off. The effects will just go totally WTF. So the death fuckup is probably just a symptom of a much bigger underlying issue.

Oh, well. This is probably already the 3rd time modders have had this EXACT problem.

I know this problem very well from other mods. The reason is a design flaw: LOTT has not any mechanism or system to properly handle status effects, especially conflicts. I assume LOTT just blindly overwrites the raw player speed/jump/gravity/etc. whenever a potion effect starts or ends. If this is the case, it is very, very wrong. Even just two contradictionary potions in quick succession can cause chaos quickly.

I have identified this problem a very long time ago, motivated by a potion mod which had the exact same problem. I then created a mod for managing status effects. It is called `playereffects`. This mod allows you to register status effects to any player, and start and stop them. The most important part in this mod is that it prevents any conflicting effects by cancelling the contradictionary effect(s), if neccessary. This mod also has the option to automatically cancel effects on death.

As a proof-of-concept, I wrote a mod called `pep` which adds some example potions. This mod is mostly just to show that `playereffects` works. You can drink potions in `pep` totally random and like crazy, the game will never mess up.

Opinions about my mod seem to be mixed. Some players do miss the functionality for additive effects, that is, multiple effects stacking. If LOTT needs fancy functionality like this, sorry. `playereffect` does not provide that.

There is a similar mod to mine out there somewhere, but I haven't tested it so far. It was something about “monoids” or something and I didn't really understand it. :D

In other words, if you would make use of `playereffects`, that other mod, or even write your own framework, you will fix the big issue (not just the death issue).

**Comment by Amaz1** [comment]
I don't really know how the potion system works, @fishyWET coded and designed it.
I know he was planning on rewriting it, don't know if anything ever came of that.

**Comment by fishyWET** [comment]
I have a working version of the new potion mod since 6 months ago, which supports both addictive effects, conflicts and stopping effects.
The reason why this is not yet pushed in is because I'm still constantly rewriting the api just so it supports a wide variety of effects and usage, rather than being limited to potions. (Blame my OCD :P)

The current potion mod uses minetest.after mainly which means that it is not possible to "fix" such issues, for now till the new version comes out, the only way is to be caution when playing with potions :/

However, if anyone want to provide a temporary solution, you are more than welcomed to.

**Comment by fishyWET** [comment]
I will close this, the issue is somewhat fixed and it will take while for the new potions due to my busy schedule.
Feel free to reopen if the issue still somehow occurs.
